src/platform/silabs/DiagnosticDataProviderImpl.cpp

```diff
@@ -52,31 +52,46 @@ DiagnosticDataProviderImpl & DiagnosticDataProviderImpl::GetDefaultInstance()
  */
 CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapFree(uint64_t & currentHeapFree)
 {
+#if (defined(SLI_SI91X_MCU_INTERFACE) && SLI_SI91X_MCU_INTERFACE == 1)
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
+#else
     size_t freeHeapSize = sl_memory_get_free_heap_size();
     currentHeapFree     = static_cast<uint64_t>(freeHeapSize);
+#endif //(defined(SLI_SI91X_MCU_INTERFACE) && SLI_SI91X_MCU_INTERFACE == 1)
     return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetCurrentHeapUsed(uint64_t & currentHeapUsed)
 {
+#if (defined(SLI_SI91X_MCU_INTERFACE) && SLI_SI91X_MCU_INTERFACE == 1)
+    return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE;
```

Comment:
Without it being sl_memory_.. Is there no API to track heap usage in wifi_sdk?

Reply:
No Junior, memory manager support is not there for 917 SoC, I have created a ticket - https://jira.silabs.com/browse/SI91X-18464